controllers/update_pods.go

```diff
@@ -87,13 +88,26 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 func getPodsToUpdate(logger logr.Logger, reconciler *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster, podMap map[fdbv1beta2.ProcessGroupID]*corev1.Pod) (map[string][]*corev1.Pod, error) {
 	updates := make(map[string][]*corev1.Pod)
 
+	var unavailablePods int
+	// When maxUnavailablePods is set to 0 any number of unavailable Pods is allowed.
+	maxUnavailablePods, err := intstr.GetScaledValueFromIntOrPercent(&cluster.Spec.MaxUnavailablePods, len(cluster.Status.ProcessGroups), true)
```

Comment:
I believe we should count Pods that have a deletionTimestamp !=0 as unavailable, what do you think?